File: src/main/kotlin/xyz/aeolia/ashutils/command/admin/ashutils/ClearChatHandler.java
package xyz.aeolia.ashutils.command.admin.ashutils;

import org.bukkit.entity.Player;
import org.bukkit.plugin.java.JavaPlugin;
import xyz.aeolia.ashutils.sender.MessageSender;

public class ClearChatHandler {
  public static boolean doClearChat(JavaPlugin plugin) {
    for (Player player : plugin.getServer().getOnlinePlayers()) {
      if (!player.hasPermission("ashutils.clearchat.exempt")) {
        for (int i = 0; i < 100; i++) {
          player.sendMessage(" ");
        }
      } else {
        MessageSender.sendMessage(player, "Chat cleared for non-exempt users.", true);
      }
    }
    return true;
  }
}

File: src/main/kotlin/xyz/aeolia/ashutils/command/admin/ashutils/ClearChatHandler.kt
package xyz.aeolia.ashutils.command.admin.ashutils

import net.kyori.adventure.text.Component
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin
import xyz.aeolia.ashutils.sender.MessageSender

class ClearChatHandler {
  companion object {
    @JvmStatic
    fun doClearChat() : Boolean {
      for (player in Bukkit.getOnlinePlayers()) {
        if(!player.hasPermission("ashutils.clearchat.exempt")) {
          repeat (100) {
            player.sendMessage(Component.text(" "))
          }
        } else {
          MessageSender.sendMessage(player, "Chat cleared for non-exempt users.")
        }
      }
      return true
    }
  }
}